src/main/java/org/fcrepo/transform/transformations/LDPathTransform.java

```diff
@@ -78,60 +79,38 @@ public LDPathTransform(final InputStream query) {
         this.query = query;
     }
 
-    /**
-     * Pull a node-type specific transform out of JCR
-     * @param node the node
-     * @param key the key
-     * @return node-type specific transform
-     * @throws RepositoryException if repository exception occurred
-     */
-    public static LDPathTransform getNodeTypeTransform(final Node node,
-        final String key) throws RepositoryException {
-
-        final Node programNode = node.getSession().getNode(CONFIGURATION_FOLDER + key);
-
-        LOGGER.debug("Found program node: {}", programNode.getPath());
-
-        final NodeType primaryNodeType = node.getPrimaryNodeType();
-
-        final Set<NodeType> supertypes = orderedBy(BY_NAME).add(primaryNodeType.getSupertypes()).build();
-        final Set<NodeType> mixinTypes = orderedBy(BY_NAME).add(node.getMixinNodeTypes()).build();
-
-        // start with mixins, primary type, and supertypes of primary type
-        final ImmutableList.Builder<NodeType> nodeTypesB = builder();
-        nodeTypesB.addAll(mixinTypes).add(primaryNodeType).addAll(supertypes);
-
-        // add supertypes of mixins
-        mixinTypes.stream().map(mixin -> orderedBy(BY_NAME).add(mixin.getDeclaredSupertypes()).build())
-            .forEach(nodeTypesB::addAll);
-
-        final List<NodeType> nodeTypes = nodeTypesB.build();
-
-        LOGGER.debug("Discovered node types: {}", nodeTypes);
-        for (final NodeType nodeType : nodeTypes) {
-            if (programNode.hasNode(nodeType.toString())) {
-                return new LDPathTransform(programNode.getNode(nodeType.toString())
-                                               .getNode(JCR_CONTENT)
-                                               .getProperty(JCR_DATA)
-                                               .getBinary().getStream());
-            }
-        }
-
-        throw new WebApplicationException(new Exception(
-                "Couldn't find transformation for " + node.getPath()
-                        + " and transformation key " + key), SC_BAD_REQUEST);
+    public static LDPathTransform getResourceTransform(final FedoraResource resource, final NodeService nodeService,
+            final String key) throws RepositoryException {
+
+        final FedoraResource programResource = nodeService.find(resource.getNode().getSession(), CONFIGURATION_FOLDER + key);
+    
+        LOGGER.debug("Found program resource: {}", programResource.getPath());
+    
+        final List<URI> rdfTypes = resource.getTypes();
+    
+        LOGGER.debug("Discovered rdf types: {}", rdfTypes);
+        
+        final List<String> rdfStringTypes = rdfTypes.stream().map(type -> programResource.getPath() + "/" + type.toString().replace( REPOSITORY_NAMESPACE, "fedora:") + "/jcr:content").collect(Collectors.toList());
```

Comment:
This is where switching from Nodes to FedoraResources got funky. 

Using Nodes it was a straight comparison of childNode.getPath() to the node types/supertypes and mixin types. Mostly because you seem to only get the relative section of the path. (ie. `nt:base`)

With Resources when you get `resource.getPath()` you get the full path (`/fedora:system/fedora:transform/default/fedora:Resource/jcr:content`) and because it is a binary it has '/jcr:content' on the end.

So to find out if there is a transform, you list the children of the transform type (ie. `default`) and match `rdf:type` of the resource you are transforming to the names of the children. If one matches, use the binary content to build the LDPathTransformation.

To answer your question @acoburn, yes I would think you should be able to add a transform for any type you can add to the repo. (ie. `pcdm:Object`) 

I'll look at the namespaceRegistry.